Higher Order Flux Corrections + Tensor Parities

acwen11 opened the pull request.

jaykalinani commented:
Thanks for looking into this, Allen!

Following are some comments:
- The higher order (HO) correction code already exists in branch `reconstruct`, which I had implemented. Could you explain why this was not used directly, and the code was copied over? If the code is copied, I suggest you correctly reference it in the PR description.
- If there was a performance issue with the old one, could you quantify what was the gain with the new re-structuring? Maybe specify which tests were run, and what % gain was obtained with the old vs new code. If there are no performance gains, I suggest keeping the original version.
- The usage of the function name `higher_order_correction` can be misleading. The default case does not apply HO corrections, yet it seems like `dfluxes` is computed by this function. I suggest either to change the name of this function (to e.g., `compute_flux_derivatives`) or restructure the code such that the applied operation is easy to read.
- Is the code giving correct physical results? I remember there was a secular trend noticed in max-density in the TOV cases. If/when the correctness is verified, I suggest adding two testsuite cases with magTOV (similar to the one that already exists), but using HO4 and HO6 methods.

- Regarding addition of parities, how well has this been tested? Is the code giving exact same results when compared with full 3D simulations? If yes, I suggest adding additional magTOV test cases, possibly one with octant and another with z-symmetry.

Thanks for all the efforts!

acwen11 commented:
Ok, I've decided to keep the same PR but revert the parities code and save that PR for another day. I think I've addressed most of the comments related to the flux code. Will try a 10 ms simulation and add tests if that works.
